org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/CompilationUnitResolver.java

```diff
@@ -81,6 +81,100 @@
 
 @SuppressWarnings({ "rawtypes", "unchecked" })
 class CompilationUnitResolver extends Compiler {
+	
+	public static final ICompilationUnitResolver FACADE = new ICompilationUnitResolver() {
```

Comment:
better a named Class to get more meaningfull stacktraces

Reply:
I'm thinking about renaming the ICompilationUnitResolver interface to ICompilationUnitResolverFacade, and the particular implementation here as `ECJCompilationUnitResolverFacade`. What do you think?

Reply:
I don't care about the actual name much. Only find it hard to read annonymous classes. Personally i would skip the postfix "Fasade", as it should be an implementation details if it's implemented as Fasace.